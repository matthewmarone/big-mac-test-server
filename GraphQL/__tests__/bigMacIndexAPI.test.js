/**
 * @jest-environment node
 */

const query = require("./test_modules/testClientAndServer");
// eslint-disable-next-line no-unused-vars
const prettyFormat = require("pretty-format");
const {
  listLatestBigMacIndex,
  listSupportedCountries,
  getLatestBigMacIndex,
} = require("./test_modules/query");

/**
 * There are no variables that this query takes, so this is the only
 * test for listLatestBigMacIndex
 */
test("Testing GraphQL query listLatestBigMacIndex", async () => {
  const { errors, data } = await query({
    query: listLatestBigMacIndex,
  });
  
  expect(errors).toBeUndefined();
  expect(data).toMatchSnapshot();
});
/**
 * There are no variables that this query takes, so this is the only
 * test for listSupportedCountries
 */
test("Testing GraphQL query listSupportedCountries", async () => {
  const { errors, data } = await query({
    query: listSupportedCountries,
  });
  expect(errors).toBeUndefined();
  expect(data).toMatchSnapshot();
});

// What follows are all the test cases for getLatestBigMacIndex

test("Testing GraphQL query getLatestBigMacIndex", async () => {
  const { errors, data } = await query({
    query: getLatestBigMacIndex,
    variables: { country: "Venezuela" },
  });

  expect(errors).toBeUndefined();
  expect(data).toMatchSnapshot();
});

test("Testing GraphQL query getLatestBigMacIndex, with unsupported country", async () => {
  const { errors, data } = await query({
    query: getLatestBigMacIndex,
    variables: { country: "Wakanda" },
  });

  expect(errors).toBeUndefined();
  expect(data).toMatchSnapshot();
});

test("Testing GraphQL query getLatestBigMacIndex, with empty String", async () => {
  const { errors, data } = await query({
    query: getLatestBigMacIndex,
    variables: { country: "" },
  });

  expect(errors).toBeUndefined();
  expect(data).toMatchSnapshot();
});

// The next two requests would be blocked by the client before going to the server
// because it violates the schema.  Country is required.

test("Testing GraphQL query getLatestBigMacIndex, with null country", async () => {
  const { errors, data } = await query({
    query: getLatestBigMacIndex,
    variables: { country: null },
  });

  expect(errors).toBeDefined();
  expect(data).toBeUndefined(); // Becasue it violates the schema
});

test("Testing GraphQL query getLatestBigMacIndex, with undefined country", async () => {
  const { errors, data } = await query({
    query: getLatestBigMacIndex,
    variables: { country: undefined },
  });

  expect(errors).toBeDefined();
  expect(data).toBeUndefined(); // Becasue it violates the schema
});
