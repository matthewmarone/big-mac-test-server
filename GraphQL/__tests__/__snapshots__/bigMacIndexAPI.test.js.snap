// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GraphQL query getLatestBigMacIndex 1`] = `
Object {
  "getLatestBigMacIndex": Object {
    "country": "Venezuela",
    "date": 1451624400,
    "dollarExchange": 198.6986,
    "dollarPPP": 26.77484787,
    "dollarPrice": 0.664322748,
    "dollarValuation": -86.52489355,
    "id": "Venezuela#LATEST",
    "localPrice": 132,
  },
}
`;

exports[`Testing GraphQL query getLatestBigMacIndex, with empty String 1`] = `
Object {
  "getLatestBigMacIndex": null,
}
`;

exports[`Testing GraphQL query getLatestBigMacIndex, with unsupported country 1`] = `
Object {
  "getLatestBigMacIndex": null,
}
`;

exports[`Testing GraphQL query listLatestBigMacIndex 1`] = `
Object {
  "listLatestBigMacIndex": Array [
    Object {
      "country": "Argentina",
      "date": 1451624400,
      "dollarExchange": 13.80925,
      "dollarPPP": 6.693711968,
      "dollarPrice": 2.389702554,
      "dollarValuation": -51.52733155,
      "id": "Argentina#LATEST",
      "localPrice": 33,
    },
    Object {
      "country": "Australia",
      "date": 1451624400,
      "dollarExchange": 1.415728746,
      "dollarPPP": 1.07505071,
      "dollarPrice": 3.743655,
      "dollarValuation": -24.0637931,
      "id": "Australia#LATEST",
      "localPrice": 5.3,
    },
    Object {
      "country": "Austria",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.709939148,
      "dollarPrice": 3.7625,
      "dollarValuation": -23.68154158,
      "id": "Austria#LATEST",
      "localPrice": 3.5,
    },
    Object {
      "country": "Belgium",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.801217039,
      "dollarPrice": 4.24625,
      "dollarValuation": -13.86916836,
      "id": "Belgium#LATEST",
      "localPrice": 3.95,
    },
    Object {
      "country": "Brazil",
      "date": 1451624400,
      "dollarExchange": 4.0248,
      "dollarPPP": 2.738336714,
      "dollarPrice": 3.354203936,
      "dollarValuation": -31.96340901,
      "id": "Brazil#LATEST",
      "localPrice": 13.5,
    },
    Object {
      "country": "Britain",
      "date": 1451624400,
      "dollarExchange": 0.684532977,
      "dollarPPP": 0.586206897,
      "dollarPrice": 4.2218565,
      "dollarValuation": -14.36396552,
      "id": "Britain#LATEST",
      "localPrice": 2.89,
    },
    Object {
      "country": "Canada",
      "date": 1451624400,
      "dollarExchange": 1.4092,
      "dollarPPP": 1.184584178,
      "dollarPrice": 4.144195288,
      "dollarValuation": -15.93924365,
      "id": "Canada#LATEST",
      "localPrice": 5.84,
    },
    Object {
      "country": "Chile",
      "date": 1451624400,
      "dollarExchange": 715.215,
      "dollarPPP": 425.9634888,
      "dollarPrice": 2.93618003,
      "dollarValuation": -40.44259574,
      "id": "Chile#LATEST",
      "localPrice": 2100,
    },
    Object {
      "country": "China",
      "date": 1451624400,
      "dollarExchange": 6.5582,
      "dollarPPP": 3.569979716,
      "dollarPrice": 2.6836632,
      "dollarValuation": -45.56464097,
      "id": "China#LATEST",
      "localPrice": 17.6,
    },
    Object {
      "country": "Colombia",
      "date": 1451624400,
      "dollarExchange": 3253.9,
      "dollarPPP": 1602.434077,
      "dollarPrice": 2.427855804,
      "dollarValuation": -50.75343197,
      "id": "Colombia#LATEST",
      "localPrice": 7900,
    },
    Object {
      "country": "Costa Rica",
      "date": 1451624400,
      "dollarExchange": 535.25,
      "dollarPPP": 436.1054767,
      "dollarPrice": 4.016814573,
      "dollarValuation": -18.52303098,
      "id": "Costa Rica#LATEST",
      "localPrice": 2150,
    },
    Object {
      "country": "Czech Republic",
      "date": 1451624400,
      "dollarExchange": 25.1442,
      "dollarPPP": 15.21298174,
      "dollarPrice": 2.982795237,
      "dollarValuation": -39.49705401,
      "id": "Czech Republic#LATEST",
      "localPrice": 75,
    },
    Object {
      "country": "Denmark",
      "date": 1451624400,
      "dollarExchange": 6.9405,
      "dollarPPP": 6.085192698,
      "dollarPrice": 4.322455155,
      "dollarValuation": -12.32342486,
      "id": "Denmark#LATEST",
      "localPrice": 30,
    },
    Object {
      "country": "Egypt",
      "date": 1451624400,
      "dollarExchange": 7.83005,
      "dollarPPP": 3.434077079,
      "dollarPrice": 2.162182872,
      "dollarValuation": -56.14233525,
      "id": "Egypt#LATEST",
      "localPrice": 16.93,
    },
    Object {
      "country": "Estonia",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.60851927,
      "dollarPrice": 3.225,
      "dollarValuation": -34.5841785,
      "id": "Estonia#LATEST",
      "localPrice": 3,
    },
    Object {
      "country": "Euro area",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.754563895,
      "dollarPrice": 3.999,
      "dollarValuation": -18.88438134,
      "id": "Euro area#LATEST",
      "localPrice": 3.72,
    },
    Object {
      "country": "Finland",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.831643002,
      "dollarPrice": 4.4075,
      "dollarValuation": -10.59837728,
      "id": "Finland#LATEST",
      "localPrice": 4.1,
    },
    Object {
      "country": "France",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.831643002,
      "dollarPrice": 4.4075,
      "dollarValuation": -10.59837728,
      "id": "France#LATEST",
      "localPrice": 4.1,
    },
    Object {
      "country": "Germany",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.728194726,
      "dollarPrice": 3.85925,
      "dollarValuation": -21.71906694,
      "id": "Germany#LATEST",
      "localPrice": 3.59,
    },
    Object {
      "country": "Greece",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.679513185,
      "dollarPrice": 3.60125,
      "dollarValuation": -26.95233266,
      "id": "Greece#LATEST",
      "localPrice": 3.35,
    },
    Object {
      "country": "Hong Kong",
      "date": 1451624400,
      "dollarExchange": 7.75235,
      "dollarPPP": 3.894523327,
      "dollarPrice": 2.476668365,
      "dollarValuation": -49.76331917,
      "id": "Hong Kong#LATEST",
      "localPrice": 19.2,
    },
    Object {
      "country": "Hungary",
      "date": 1451624400,
      "dollarExchange": 292.67905,
      "dollarPPP": 182.5557809,
      "dollarPrice": 3.075040731,
      "dollarValuation": -37.62594865,
      "id": "Hungary#LATEST",
      "localPrice": 900,
    },
    Object {
      "country": "India",
      "date": 1451624400,
      "dollarExchange": 66.8025,
      "dollarPPP": 25.76064909,
      "dollarPrice": 1.901126455,
      "dollarValuation": -61.43759726,
      "id": "India#LATEST",
      "localPrice": 127,
    },
    Object {
      "country": "Indonesia",
      "date": 1451624400,
      "dollarExchange": 13947.5,
      "dollarPPP": 6186.612576,
      "dollarPrice": 2.186771823,
      "dollarValuation": -55.64357357,
      "id": "Indonesia#LATEST",
      "localPrice": 30500,
    },
    Object {
      "country": "Ireland",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.801217039,
      "dollarPrice": 4.24625,
      "dollarValuation": -13.86916836,
      "id": "Ireland#LATEST",
      "localPrice": 3.95,
    },
    Object {
      "country": "Israel",
      "date": 1451624400,
      "dollarExchange": 3.94105,
      "dollarPPP": 3.427991886,
      "dollarPrice": 4.288197308,
      "dollarValuation": -13.01831019,
      "id": "Israel#LATEST",
      "localPrice": 16.9,
    },
    Object {
      "country": "Italy",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.811359026,
      "dollarPrice": 4.3,
      "dollarValuation": -12.77890467,
      "id": "Italy#LATEST",
      "localPrice": 4,
    },
    Object {
      "country": "Japan",
      "date": 1451624400,
      "dollarExchange": 118.645,
      "dollarPPP": 75.05070994,
      "dollarPrice": 3.118546926,
      "dollarValuation": -36.74347007,
      "id": "Japan#LATEST",
      "localPrice": 370,
    },
    Object {
      "country": "Latvia",
      "date": 1372654800,
      "dollarExchange": 0.54635,
      "dollarPPP": 0.37088515,
      "dollarPrice": 3.093255239,
      "dollarValuation": -32.11583235,
      "id": "Latvia#LATEST",
      "localPrice": 1.69,
    },
    Object {
      "country": "Lithuania",
      "date": 1404190800,
      "dollarExchange": 2.56455,
      "dollarPPP": 1.866527633,
      "dollarPrice": 3.489891014,
      "dollarValuation": -27.21812275,
      "id": "Lithuania#LATEST",
      "localPrice": 8.95,
    },
    Object {
      "country": "Malaysia",
      "date": 1451624400,
      "dollarExchange": 4.3925,
      "dollarPPP": 1.622718053,
      "dollarPrice": 1.821286283,
      "dollarValuation": -63.05707336,
      "id": "Malaysia#LATEST",
      "localPrice": 8,
    },
    Object {
      "country": "Mexico",
      "date": 1451624400,
      "dollarExchange": 17.43675,
      "dollarPPP": 9.939148073,
      "dollarPrice": 2.810156709,
      "dollarValuation": -42.99884971,
      "id": "Mexico#LATEST",
      "localPrice": 49,
    },
    Object {
      "country": "Netherlands",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.69979716,
      "dollarPrice": 3.70875,
      "dollarValuation": -24.77180527,
      "id": "Netherlands#LATEST",
      "localPrice": 3.45,
    },
    Object {
      "country": "New Zealand",
      "date": 1451624400,
      "dollarExchange": 1.508068165,
      "dollarPPP": 1.196754564,
      "dollarPrice": 3.91229,
      "dollarValuation": -20.64320487,
      "id": "New Zealand#LATEST",
      "localPrice": 5.9,
    },
    Object {
      "country": "Norway",
      "date": 1451624400,
      "dollarExchange": 8.97465,
      "dollarPPP": 9.492900609,
      "dollarPrice": 5.214688038,
      "dollarValuation": 5.774605233,
      "id": "Norway#LATEST",
      "localPrice": 46.8,
    },
    Object {
      "country": "Pakistan",
      "date": 1451624400,
      "dollarExchange": 104.885,
      "dollarPPP": 60.85192698,
      "dollarPrice": 2.86027554,
      "dollarValuation": -41.98224057,
      "id": "Pakistan#LATEST",
      "localPrice": 300,
    },
    Object {
      "country": "Peru",
      "date": 1451624400,
      "dollarExchange": 3.418,
      "dollarPPP": 2.028397566,
      "dollarPrice": 2.925687537,
      "dollarValuation": -40.65542522,
      "id": "Peru#LATEST",
      "localPrice": 10,
    },
    Object {
      "country": "Philippines",
      "date": 1451624400,
      "dollarExchange": 47.0225,
      "dollarPPP": 26.57200811,
      "dollarPrice": 2.785900367,
      "dollarValuation": -43.49086477,
      "id": "Philippines#LATEST",
      "localPrice": 131,
    },
    Object {
      "country": "Poland",
      "date": 1451624400,
      "dollarExchange": 4.0477,
      "dollarPPP": 1.947261663,
      "dollarPrice": 2.371717272,
      "dollarValuation": -51.89214459,
      "id": "Poland#LATEST",
      "localPrice": 9.6,
    },
    Object {
      "country": "Portugal",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.60851927,
      "dollarPrice": 3.225,
      "dollarValuation": -34.5841785,
      "id": "Portugal#LATEST",
      "localPrice": 3,
    },
    Object {
      "country": "Russia",
      "date": 1451624400,
      "dollarExchange": 74.655,
      "dollarPPP": 23.12373225,
      "dollarPrice": 1.527024312,
      "dollarValuation": -69.02587603,
      "id": "Russia#LATEST",
      "localPrice": 114,
    },
    Object {
      "country": "Saudi Arabia",
      "date": 1451624400,
      "dollarExchange": 3.7543,
      "dollarPPP": 2.434077079,
      "dollarPrice": 3.196334869,
      "dollarValuation": -35.16562131,
      "id": "Saudi Arabia#LATEST",
      "localPrice": 12,
    },
    Object {
      "country": "Singapore",
      "date": 1451624400,
      "dollarExchange": 1.43725,
      "dollarPPP": 0.953346856,
      "dollarPrice": 3.270133936,
      "dollarValuation": -33.66868283,
      "id": "Singapore#LATEST",
      "localPrice": 4.7,
    },
    Object {
      "country": "South Africa",
      "date": 1451624400,
      "dollarExchange": 15.81375,
      "dollarPPP": 5.679513185,
      "dollarPrice": 1.770611019,
      "dollarValuation": -64.08496919,
      "id": "South Africa#LATEST",
      "localPrice": 28,
    },
    Object {
      "country": "South Korea",
      "date": 1451624400,
      "dollarExchange": 1197.75,
      "dollarPPP": 872.2109533,
      "dollarPrice": 3.590064705,
      "dollarValuation": -27.17921492,
      "id": "South Korea#LATEST",
      "localPrice": 4300,
    },
    Object {
      "country": "Spain",
      "date": 1451624400,
      "dollarExchange": 0.930232558,
      "dollarPPP": 0.709939148,
      "dollarPrice": 3.7625,
      "dollarValuation": -23.68154158,
      "id": "Spain#LATEST",
      "localPrice": 3.5,
    },
    Object {
      "country": "Sri Lanka",
      "date": 1451624400,
      "dollarExchange": 144.05,
      "dollarPPP": 70.99391481,
      "dollarPrice": 2.429711906,
      "dollarValuation": -50.71578285,
      "id": "Sri Lanka#LATEST",
      "localPrice": 350,
    },
    Object {
      "country": "Sweden",
      "date": 1451624400,
      "dollarExchange": 8.60185,
      "dollarPPP": 9.127789047,
      "dollarPrice": 5.231432773,
      "dollarValuation": 6.114255034,
      "id": "Sweden#LATEST",
      "localPrice": 45,
    },
    Object {
      "country": "Switzerland",
      "date": 1451624400,
      "dollarExchange": 1.0088,
      "dollarPPP": 1.318458418,
      "dollarPrice": 6.443298969,
      "dollarValuation": 30.69571945,
      "id": "Switzerland#LATEST",
      "localPrice": 6.5,
    },
    Object {
      "country": "Taiwan",
      "date": 1451624400,
      "dollarExchange": 33.2315,
      "dollarPPP": 13.9959432,
      "dollarPrice": 2.076343229,
      "dollarValuation": -57.88350449,
      "id": "Taiwan#LATEST",
      "localPrice": 69,
    },
    Object {
      "country": "Thailand",
      "date": 1451624400,
      "dollarExchange": 36.22,
      "dollarPPP": 22.71805274,
      "dollarPrice": 3.092214246,
      "dollarValuation": -37.2776015,
      "id": "Thailand#LATEST",
      "localPrice": 112,
    },
    Object {
      "country": "Turkey",
      "date": 1451624400,
      "dollarExchange": 3.0096,
      "dollarPPP": 2.079107505,
      "dollarPrice": 3.405768208,
      "dollarValuation": -30.91748056,
      "id": "Turkey#LATEST",
      "localPrice": 10.25,
    },
    Object {
      "country": "UAE",
      "date": 1451624400,
      "dollarExchange": 3.6731,
      "dollarPPP": 2.636916836,
      "dollarPrice": 3.53924478,
      "dollarValuation": -28.21004504,
      "id": "UAE#LATEST",
      "localPrice": 13,
    },
    Object {
      "country": "Ukraine",
      "date": 1451624400,
      "dollarExchange": 23.35,
      "dollarPPP": 7.302231237,
      "dollarPrice": 1.541755889,
      "dollarValuation": -68.72706108,
      "id": "Ukraine#LATEST",
      "localPrice": 36,
    },
    Object {
      "country": "United States",
      "date": 1451624400,
      "dollarExchange": 1,
      "dollarPPP": 1,
      "dollarPrice": 4.93,
      "dollarValuation": 0,
      "id": "United States#LATEST",
      "localPrice": 4.93,
    },
    Object {
      "country": "Uruguay",
      "date": 1451624400,
      "dollarExchange": 30.185,
      "dollarPPP": 22.92089249,
      "dollarPrice": 3.743581249,
      "dollarValuation": -24.06528907,
      "id": "Uruguay#LATEST",
      "localPrice": 113,
    },
    Object {
      "country": "Venezuela",
      "date": 1451624400,
      "dollarExchange": 198.6986,
      "dollarPPP": 26.77484787,
      "dollarPrice": 0.664322748,
      "dollarValuation": -86.52489355,
      "id": "Venezuela#LATEST",
      "localPrice": 132,
    },
    Object {
      "country": "Vietnam",
      "date": 1451624400,
      "dollarExchange": 22467.5,
      "dollarPPP": 12170.3854,
      "dollarPrice": 2.67052409,
      "dollarValuation": -45.83115435,
      "id": "Vietnam#LATEST",
      "localPrice": 60000,
    },
  ],
}
`;

exports[`Testing GraphQL query listSupportedCountries 1`] = `
Object {
  "listSupportedCountries": Object {
    "countries": Array [
      "Argentina",
      "Australia",
      "Austria",
      "Belgium",
      "Brazil",
      "Britain",
      "Canada",
      "Chile",
      "China",
      "Colombia",
      "Costa Rica",
      "Czech Republic",
      "Denmark",
      "Egypt",
      "Estonia",
      "Euro area",
      "Finland",
      "France",
      "Germany",
      "Greece",
      "Hong Kong",
      "Hungary",
      "India",
      "Indonesia",
      "Ireland",
      "Israel",
      "Italy",
      "Japan",
      "Latvia",
      "Lithuania",
      "Malaysia",
      "Mexico",
      "Netherlands",
      "New Zealand",
      "Norway",
      "Pakistan",
      "Peru",
      "Philippines",
      "Poland",
      "Portugal",
      "Russia",
      "Saudi Arabia",
      "Singapore",
      "South Africa",
      "South Korea",
      "Spain",
      "Sri Lanka",
      "Sweden",
      "Switzerland",
      "Taiwan",
      "Thailand",
      "Turkey",
      "UAE",
      "Ukraine",
      "United States",
      "Uruguay",
      "Venezuela",
      "Vietnam",
    ],
    "id": "LATEST",
  },
}
`;
